nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/deck_guns.dm

```diff
@@ -15,13 +20,68 @@
 	max_ammo = 1
 	obj_integrity = 500
 	max_integrity = 500
+	component_parts = list()
 	safety = FALSE
 	maintainable = FALSE //This just makes them brick.
 	load_sound = 'nsv13/sound/effects/ship/freespace2/crane_short.ogg'
+	interaction_flags_machine = INTERACT_MACHINE_OPEN | INTERACT_MACHINE_OFFLINE | INTERACT_MACHINE_WIRES_IF_OPEN | INTERACT_MACHINE_ALLOW_SILICON | INTERACT_MACHINE_OPEN_SILICON | INTERACT_MACHINE_SET_MACHINE
 	var/obj/machinery/deck_turret/core
 	var/id = null //N.B. This is NOT intended to allow them to manual link deck guns. This is for certain boarding maps and is thus a UNIQUE CONSTRAINT for this one case. ~KMC
 	circuit = /obj/item/circuitboard/machine/deck_turret
 
+/obj/machinery/ship_weapon/deck_turret/Topic(href, href_list)
+	. = ..() //Sanity checks.
+	if(.)
+		return
+	
+	if(href_list["fire_button"])
+		if(maint_state == MSTATE_UNSCREWED)
+			fire()
+
+/obj/machinery/ship_weapon/deck_turret/examine()
+	. = ..()
+	if(maint_state == MSTATE_UNSCREWED)
+		. += "There is a button labelled \"<A href='?src=[REF(src)];fire_button=1'>Force Eject Shell</A>\"."
+	if(maint_state == MSTATE_UNBOLTED)
+		pop(.)// this is the laziest way I know of to change an examine line
+		. += "The inner casing has been <b>unbolted</b>, and the loading tray can be <i>pried out</i>."//deconstruction hint
+
+/obj/machinery/ship_weapon/deck_turret/crowbar_act(mob/user, obj/item/tool)
+	if(maint_state == MSTATE_UNBOLTED)
+		to_chat(user, "<span class='notice'>You start removing the loading tray from the [src].</span>")
+		if(!do_after(user, 4 SECONDS, target=src))
+			return
+		var/obj/item/ship_weapon/parts/loading_tray/W = locate() in component_parts
+		if(W)
+			W.forceMove(user.loc)
+			component_parts -= W
+		spawn_frame(TRUE)
+		qdel(src)
+	
+/obj/machinery/ship_weapon/deck_turret/spawn_frame(disassembled)
+	if(!disassembled)
+		for(var/obj/O in component_parts)
+			qdel(O)
+		component_parts.Cut()
+		return ..(disassembled)
+
+	circuit.moveToNullspace()//if you can't delete it...
+	circuit = null
+	var/obj/structure/ship_weapon/artillery_frame/M = new(get_turf(src))
+
+	for(var/obj/O in component_parts)
```

Comment:
empty `as()` saves processing time, it stops the loop from type checking. Useful when you know there's never going to be a different object type in the list
```suggestion
	for(var/obj/O as() in component_parts)
```